File: Homework_2/app/src/main/java/com/example/homework_2/models/Rocket.java
package com.example.homework_2.models;

public class Rocket implements SpaceShip{

   private int cost;
   private int weight;
   private int maxWeight;
   private int currentWeight;

    public boolean launch() {
        return true;
    }

    public boolean land() {
        return true;
    }

    public boolean canCarry(Item item) {
        return this.currentWeight + item.weight <= maxWeight;
    }

    public int carry(Item item) {
        return this.currentWeight += item.weight;
    }

    public int getCost() {
        return cost;
    }

    public int getCurrentWeight() {
        return currentWeight;
    }

    public int getMaxWeight() {
        return maxWeight;
    }

    public int getWeight() {
        return weight;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }

    public void setMaxWeight(int maxWeight) {
        this.maxWeight = maxWeight;
    }

    public void setCurrentWeight(int currentWeight) {
        this.currentWeight = currentWeight;
    }
}

File: SpaceShip_Kotlin/app/src/main/java/com/example/spaceship_kotlin/moduls/Rocket.kt
package com.example.spaceship_kotlin.moduls

open class Rocket : SpaceShip {
    internal var cost: Int = 0
    var weight: Int = 0
    var maxWeight: Int = 0
    var currentWeight: Int = 0

    override fun launch(): Boolean = true

    override fun land(): Boolean = true

    override fun canCarry(item: Item?): Boolean{
        return this.currentWeight + item!!.weight <= maxWeight
    }

    override fun carry(item: Item?): Int{
        this.currentWeight +=item!!.weight
        return this.currentWeight
    }

    fun getCost(): Int{ return this.cost }
}